docs/src/pages/[platform]/connected-components/storage/storage-browser/examples/GenerateUrlViewComposed.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { createStorageBrowser } from '@aws-amplify/ui-react-storage/browser';
+import { Button, Flex, Link, StepperField, Text } from '@aws-amplify/ui-react';
+import { generateUrlHandler } from './generateUrlHandler';
+import { mockConfig } from './mockConfig'; // IGNORE
+import { defaultActions } from './defaultActions'; // IGNORE
+
+const { StorageBrowser, useAction, useView } = createStorageBrowser({
+  actions: {
+    default: defaultActions, // IGNORE
+    custom: {
+      generateUrl: {
+        actionListItem: {
+          icon: 'download',
+          label: 'Generate Download Links',
+          disable: (selected) => !selected?.length,
+        },
+        handler: generateUrlHandler,
+        viewName: 'GenerateUrlView',
+      },
+    },
+  },
+  config: mockConfig, // IGNORE
+});
+
+const GenerateUrlView = () => {
+  const [duration, setDuration] = React.useState(60);
+
+  const { onActionExit, fileDataItems } = useView('LocationDetail');
+
+  const items = React.useMemo(
+    () =>
+      !fileDataItems
+        ? []
+        : fileDataItems.map((item) => ({ ...item, duration })),
+    [fileDataItems, duration]
+  );
+
+  const [
+    { tasks }, // Execution status and result of each task. The status includes  'CANCELED', 'FAILED', 'COMPLETE', 'OVERWRITE_PREVENTED', 'QUEUED', 'PENDING';
+    handleCreate, // Start executing the action at batch size of 4.
+  ] = useAction(
+    'generateUrl', // Name of the action
+    { items } // List of action inputs.
```

Comment:
Prefer placing the comments above rather than in the same line, also not sure if we need to callout the batch size since it's not configurable:
```suggestion
    // Execution status and result of each task. The status includes  'CANCELED', 'FAILED', 'COMPLETE', 'OVERWRITE_PREVENTED', 'QUEUED', 'PENDING';
    { tasks }, 
    // Start executing the action against the provided `items`
    handleCreate, 
  ] = useAction(
    // Name of the action
    'generateUrl', 
    // List of action inputs.
    { items }
```